x501/src/name.rs

```diff
@@ -0,0 +1,58 @@
+//! Name-related definitions as defined in X.501 (and updated by RFC 5280).
+
+use crate::attr::AttributeTypeAndValue;
+
+use der::asn1::SetOfVec;
+
+/// X.501 Name as defined in [RFC 5280 Section 4.1.2.4]. X.501 Name is used to represent distinguished names.
+///
+/// ```text
+/// Name ::= CHOICE { rdnSequence  RDNSequence }
+/// ```
+///
+/// [RFC 5280 Section 4.1.2.4]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4
+pub type Name<'a> = RDNSequence<'a>;
+
+/// RDNSequence as defined in [RFC 5280 Section 4.1.2.4].
+///
+/// ```text
+/// RDNSequence ::= SEQUENCE OF RelativeDistinguishedName
+/// ```
+///
+/// [RFC 5280 Section 4.1.2.4]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4
+pub type RDNSequence<'a> = alloc::vec::Vec<RelativeDistinguishedName<'a>>;
```

Comment:
Idiomatic Rust naming has generally moved to "capitalized acronyms", which is to say the first letter of an acronym is capitalized. See https://rust-lang.github.io/api-guidelines/naming.html

> In `UpperCamelCase`, acronyms and contractions of compound words count as one word: use `Uuid` rather than `UUID`, `Usize` rather than `USize` or `Stdin` rather than `StdIn`. 

```suggestion
pub type RdnSequence<'a> = alloc::vec::Vec<RelativeDistinguishedName<'a>>;
```

There are clippy lints for this on structs, but apparently not on type aliases.

Reply:
It was all caps in the previous version. So I just copied that. Fixed now.